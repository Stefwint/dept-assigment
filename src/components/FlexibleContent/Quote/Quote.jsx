import React from 'react';
import PropTypes from 'prop-types';
import theme from '../../../theme';

// Styling
import { Box, Flex, Row, Text } from '../../styles';

const Quote = ({ item }) => {
  const { quote, name } = item;

  return (
    <Row px={0} mt={40}>
      <Flex pt={80} pb={60} justifyContent="center" className="border">
        <Box width={[1, 8 / 12]}>
          {quote && (
            <Text as="blockquote" fontSize={36}>
              {quote}
            </Text>
          )}
          {name && (
            <Text
              as="span"
              mt={30}
              fontSize={13}
              fontWeight="700"
              color={theme.colors.blackTwo}
              className="block upper"
            >
              {name}
            </Text>
          )}
        </Box>
      </Flex>
    </Row>
  );
};

Quote.propTypes = {
  item: PropTypes.object.isRequired,
};

export default Quote;
