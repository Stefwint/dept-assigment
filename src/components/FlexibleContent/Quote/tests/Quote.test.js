import React from 'react';
import renderer from 'react-test-renderer';
import Quote from '../Quote';

describe('testing Quote', () => {
  const item = (
    <Quote
      item={
        {
          quote: '“Dept helped us tell our story through a bold new identity and a robust online experience. To the tune of 60% growth in online bookings in just one month.”',
          name: 'MATTIJS TEN BRINK – CEO, TRANSAVIA'
        }
      }
    />
  );

  it('Renders correctly', () => {
    const renderItem = renderer.create(item).toJSON();
    expect(renderItem).toMatchSnapshot();
  });
});
