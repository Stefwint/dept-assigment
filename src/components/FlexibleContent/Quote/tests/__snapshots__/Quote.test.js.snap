// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Quote Renders correctly 1`] = `
<div
  className="Flex__Box-ro60kk-0 Row-sc-1bf1evj-0 epzZas"
>
  <div
    className="Flex__Box-ro60kk-0 Flex-ro60kk-2 ihhDQr border"
  >
    <div
      className="Flex__Box-ro60kk-0 gyKgXO"
      width={
        Array [
          1,
          0.8333333333333334,
          0.6666666666666666,
        ]
      }
    >
      <blockquote
        className="Flex__Text-ro60kk-1 bXZXsb"
        fontSize={
          Array [
            24,
            36,
          ]
        }
      >
        “Dept helped us tell our story through a bold new identity and a robust online experience. To the tune of 60% growth in online bookings in just one month.”
      </blockquote>
      <span
        className="Flex__Text-ro60kk-1 dlCxQQ block upper"
        color="#272727"
        fontSize={13}
        fontWeight="700"
      >
        MATTIJS TEN BRINK – CEO, TRANSAVIA
      </span>
    </div>
  </div>
</div>
`;
